wp-content/civi-extensions/goonjcustom/Civi/CollectionCampHelper.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace Civi;
+
+use Civi\Core\Service\AutoSubscriber;
+
+class CollectionCampHelper extends AutoSubscriber {
+
+	public static function getSubscribedEvents() {
+		return [
+			'&hook_civicrm_post' => 'generateCollectionCampCode',
+		];
+	}
+
+	/**
+	 * This hook is called after a db write on entities.
+	 *
+	 * @param string $op
+	 *   The type of operation being performed.
+	 * @param string $objectName
+	 *   The name of the object.
+	 * @param int $objectId
+	 *   The unique identifier for the object.
+	 * @param object $objectRef
+	 *   The reference to the object.
+	 */
+	public static function generateCollectionCampCode(string $op, string $objectName, int $objectId, &$objectRef) {
+		// Check if the object name is 'AfformSubmission'
+		if ($objectName !== 'AfformSubmission') {
+			return;
+		}
+	
+		// Extract the 'data' field
+		$data = $objectRef->data;
+		$decodedData = json_decode($data, true);
+	
+		// Check if 'Eck_Collection_Camp1' exists
+		$collectionCampEntries = $decodedData['Eck_Collection_Camp1'] ?? [];
+		if (empty($collectionCampEntries)) {
+			return;
+		}
+	
+		foreach ($collectionCampEntries as $entry) {
+			$collectionCampData = $entry['fields'] ?? null;
+			if (!array_key_exists('Eck_Collection_Camp1', $decodedData)) {
+				continue;
+			}
+	
+			// Access the subtype
+			$subtypeId = $collectionCampData['subtype'] ?? null;
+			if ($subtypeId === null) {
+				continue;
+			}
+	
+			// Access the id within the decoded data
+			$campId = $collectionCampData['id'] ?? null;
+			if ($campId === null) {
+				continue;
+			}
+	
+			// Fetch the collection camp details
+			$collectionCamps = \Civi\Api4\EckEntity::get('Collection_Camp', FALSE)
+				->addWhere('id', '=', $campId)
+				->setLimit(1)
+				->execute();
+	
+			if (empty($collectionCamps)) {
+				continue;
+			}
+	
+			$collectionCampsCreatedDate = $collectionCamps->first()['created_date'] ?? null;
+	
+			// Get the year
+			$year = date('Y', strtotime($collectionCampsCreatedDate));
+	
+			// Fetch the state ID from the collection camp intent details
+			$stateId = $collectionCampData['Collection_Camp_Intent_Details.State'] ?? null;
+			if ($subtypeId == 5) {
+				$stateId = $collectionCampData['Dropping_Centre.State'] ?? null;
+			}
+	
+			if (!$stateId) {
+				continue;
+			}
```

Comment:
Move this to a separate function
```suggestion
			// Fetch the state ID from the collection camp intent details
			$stateId = $this->getStateId($collectionCampData, $subType);
```